scipy/sparse/_coo.py

```diff
@@ -1367,6 +1808,34 @@ def _ravel_non_reduced_axes(coords, shape, axes):
     return np.ravel_multi_index(non_reduced_coords, non_reduced_shape)
 
 
+def _validateaxes(axis, ndim, shape):
```

Comment:
Let's see if we can merge this with the original validateaxes in sputils, maybe using a kwarg like `allow_nd` to gate the new behavior.